pyproject.toml

```diff
@@ -46,3 +48,17 @@ sphinx-rtd-theme = ">=1.0"
 [build-system]
 requires = ["poetry-core"]
 build-backend = "poetry.core.masonry.api"
+
+[tool.mypy]
+show_error_codes = true
+strict = false
```

Comment:
note currently this is not strict type checking, eventually you would want to set this to true, but currently there are lots of errors